deps/src/ideals.cpp

```diff
@@ -677,6 +677,22 @@ void singular_define_ideals(jlcxx::Module & Singular)
         delete v;
         rChangeCurrRing(origin);
     });
+    Singular.method("scHilbPoly", [](ideal I, ring r, ring Qt) {
+        const ring origin = currRing;
+        rChangeCurrRing(r);
+        poly h=hFirstSeries0p(I,r->qideal,NULL,r,Qt);
+        rChangeCurrRing(origin);
+	return h;
+    });
+    Singular.method("scHilbPolyWeighted", [](ideal I, ring r, jlcxx::ArrayRef<int> weights, ring Qt) {
+        intvec *w = to_intvec(weights);
```

Comment:
Assumes that `weights` has the correct size, and that all its entries are positive (and maybe not too big?)